contrib/auctex/packages.el

```diff
@@ -12,36 +13,36 @@
     :defer t
     :config
     (progn
-      (defun auctex/build-view ()
-        (interactive)
-        (progn
-          (let ((TeX-save-query nil))
-            (TeX-save-document (TeX-master-file)))
-          (setq build-proc (TeX-command "LaTeX" 'TeX-master-file -1))
-          (set-process-sentinel  build-proc  'auctex/build-sentinel))
-        (TeX-view))
-
-      (defun auctex/build-sentinel (process event)
-        (if (string= event "finished\n")
-            (TeX-view)
-          (message "Errors! Check with C-`")))
-
-      (add-hook 'LaTeX-mode-hook '(lambda () (local-set-key (kbd "H-r") 'auctex/build-view)))
-      (add-hook 'LaTeX-mode-hook 'flyspell-mode)
-      (add-hook 'LaTeX-mode-hook 'company-mode)
-      (add-hook 'LaTeX-mode-hook 'LaTeX-math-mode)
-      (add-hook 'LaTeX-mode-hook 'spacemacs/load-yasnippet)
-
-      (setq spacemacs/key-binding-prefixes '(("mp" . "LaTeX Preview")))
+      (add-hook 'TeX-mode-hook 'flyspell-mode)
+      (add-hook 'TeX-mode-hook 'company-mode)
+      (add-hook 'TeX-mode-hook 'spacemacs/load-yasnippet)
+      (when auctex-auto-fill (add-hook 'TeX-mode-hook 'auctex/auto-fill-mode))
+      (add-hook 'TeX-mode-hook 'smartparens-mode)
```

Comment:
flyspell, company, yasnippet and smartparens hooks should be added in a corresponding `auctex/post-init-<package>` where `package` is flyspell, company etc...
This is required to keep the modularity of spacemacs layer, if a user exclude flyspell package for some reason then the hook should not be added.